addons/hr_holidays/models/hr_employee.py

```diff
@@ -387,7 +387,7 @@ def _get_consumed_leaves(self, leave_types, target_date=False, ignore_future=Fal
         leaves_domain = [
             ('holiday_status_id', 'in', leave_types.ids),
             ('employee_id', 'in', employees.ids),
-            ('state', 'in', ['confirm', 'validate1', 'validate']),
+            ('state', 'in', ['confirm', 'validate1', 'validate', 'draft']),
```

Comment:
This isn't the right way to fix it; draft leaves aren't suppose to be taken into account when computing the balance of the employee. Instead, you'll probably need to check the validity by adding the state field here
`https://github.com/odoo/odoo/blob/17.0/addons/hr_holidays/models/hr_leave.py#L999`